### PR TITLE
Add add-auth attribute for manifold-marketplace

### PR DESCRIPTION
**Pull request by dangodev**
## Reason for change

For components that use catalog, it’s difficult to determine whether or not they need to wait for auth. Since most consumers **aren’t** using it, let’s do that by default, but allow a `use-auth` prop that gets passed in for cases they need it. 

This was based on some discussions we had, and the logic around when to wait (or not) was overly complex. But by passing the responsibility to the consumer on what to use, we can be more certain we’re handling that usecase. As a result we can simplify the auth code in UI.

## Testing

- [ ] In Storybook, `manifold-marketplace` doesn’t use auth by default
- [ ] (eventually, we’ll remove the `catalog` check in restFetch so for now the “with auth“ story doesn’t authenticate, but the plan is once all these components use this, remove the catalog-specific check)

## Checklist

- [x] **CHANGELOG**: The **Unreleased** section of CHANGELOG was updated
- [x] **Prop changes**: [docs][docs] were updated
- [x] **Prop changes**: E2E tests were updated, testing from the highest level possible

[docs]: https://ui.sandbox.manifold.co

### CODE REVIEW

File: src/components/manifold-marketplace/manifold-marketplace.tsx [hunk 1]
```diff
@@ -25,6 +25,8 @@ export class ManifoldMarketplace {
   @Prop() productLinkFormat?: string;
   /** Comma-separated list of shown products (labels) */
   @Prop() products?: string;
+  /** Use auth? */
+  @Prop() useAuth?: boolean = false;
```
Comment (davidleger95) [inline src/components/manifold-marketplace/manifold-marketplace.tsx]: is there any reason why we wouldn't always request without auth?

Only thing that comes to mind controlling the products that can be accessed by a user? or maybe a platform? but platform access is going to be handled by a different header, right?

File: docs/docs/components/manifold-marketplace.md [hunk 2]
```diff
@@ -98,3 +96,15 @@ also be turned into an `<a>` tag by specifying `product-link-format` and `templa
...
 
 Note that this will disable the custom events unless `preserve-event` is passed as well.
+
+## Authentication
+
+By default, `<manifold-marketplace>` is a public component. However, when used in conjunction with
+[Authentication][auth], you can request to authenticate with `with-auth`:
+
+```html
+<manifold-marketplace with-auth></manifold-marketplace>
```
Comment (davidleger95) [inline docs/docs/components/manifold-marketplace.md]: is `with-auth` the naming we want to use? Personally that naming I associate with wrapping a component in a function that injects props. like I'd expect: `withAuth(ManifoldMarketplace)`

alternative suggestion: `auth-required`

File: src/components/manifold-marketplace/manifold-marketplace.tsx (hunk 1 above)
Comment (sslotsky) [reply src/components/manifold-marketplace/manifold-marketplace.tsx]: As I understand it, platforms may want to see certain hidden products and plans. For that, wouldn't they need a platform token?

Comment (dangodev) [reply src/components/manifold-marketplace/manifold-marketplace.tsx]: You do have a point—at the moment I’m not aware of any actual need for platforms to access hidden products. You’re right in that they will have a specific list curated for them, but none of it uses auth.

I think only times platforms would need auth would be in the provider role, when they’re a platform and they are also releasing services. But to Dave’s point, we’re just anticipating needs, and carrying through the `use-auth` work would be a lot of work now for mere anticipation.

Removing all auth for catalog now, and making this change later, would solve the original problem I was trying to solve (reduce the complexity around the wait/don’t wait logic), and be simple to change later.

File: docs/docs/components/manifold-marketplace.md (hunk 2 above)
Comment (dangodev) [reply docs/docs/components/manifold-marketplace.md]: Ah this was a typo. Was supposed to be `use-auth`

File: src/components/manifold-marketplace/manifold-marketplace.tsx (hunk 1 above)
Comment (sslotsky) [reply src/components/manifold-marketplace/manifold-marketplace.tsx]: I thought Digital Ocean would need this, but I'm not opposed to waiting.

Comment (dangodev) [reply src/components/manifold-marketplace/manifold-marketplace.tsx]: Cool. In that case, I’ll close this PR, and we’ll just count on **always** skipping auth for catalog in web components (and remove the TODOs basically).

**Comment by dangodev** [comment]
Closing; let’s keep all catalog requests public for now and revisit when needed (hopefully we’ll be on GraphQL by then)

**Pull request closed without merging**